### PR TITLE
backup describer: show snapshot summary by default, details optionally

### LINKED ISSUE #973: [v0.10] update `ark backup describe` for the changes to volume snapshot info storage
With #922, information about volume snapshots is no longer stored as part of the Backup CRD.  The `ark backup describe` command used to display this information directly from the status.  It needs to be updated to work with the new design.  My proposal:

- [ ] add 2 new fields to `Backup`'s status, `VolumeSnapshotsAttempted` and `VolumeSnapshotsSucceeded`
- [ ] by default, `ark backup describe` will only display this summary information (e.g. `2 of 3 volume snapshots completed successfully`)
- [ ] change the current `--volume-details` flag on `ark backup describe` (used to show details of restic snapshots) to a generic `--details` flag
- [ ] add a new `DownloadRequest` target type, for the `<backup>-volumesnapshots.json.gz` file in object storage
- [ ] if the `--details` flag on `ark backup describe` is specified, download the volumesnapshots file from obj storage (similarly to how restore results are downloaded) and display the detailed info

@nrb @wwitzel3 @carlisia @ncdc WDYT?

### PR DESCRIPTION
**Pull request by skriss**
Signed-off-by: Steve Kriss <steve@heptio.com>

Fixes #973

### CODE REVIEW

File: pkg/cmd/util/output/backup_describer.go [hunk 1]
```diff
@@ -197,22 +211,49 @@ func DescribeBackupStatus(d *Describer, status arkv1api.BackupStatus) {
...
-			d.Printf("\t\tAvailability Zone:\t%s\n", info.AvailabilityZone)
-			iops := "<N/A>"
-			if info.Iops != nil {
-				iops = fmt.Sprintf("%d", *info.Iops)
+			printSnapshot(d, pvName, info.SnapshotID, info.Type, info.AvailabilityZone, info.Iops)
+		}
+	} else if status.VolumeSnapshotsAttempted > 0 {
+		// v0.10+ backup
+		if !details {
+			d.Printf("Persistent Volumes:\t%d of %d snapshots completed successfully (specify --details for more information)\n", status.VolumeSnapshotsCompleted, status.VolumeSnapshotsAttempted)
```
Comment (carlisia) [inline pkg/cmd/util/output/backup_describer.go]: I'd add a return here, then you don't need to have the `else`.

File: pkg/cmd/util/output/backup_describer.go [hunk 2]
```diff
@@ -197,22 +211,49 @@ func DescribeBackupStatus(d *Describer, status arkv1api.BackupStatus) {
...
+		} else {
+			buf := new(bytes.Buffer)
+			if err := downloadrequest.Stream(arkClient.ArkV1(), backup.Namespace, backup.Name, arkv1api.DownloadTargetKindBackupVolumeSnapshots, buf, 30*time.Second); err != nil {
```
Comment (carlisia) [inline pkg/cmd/util/output/backup_describer.go]: Should we make the `30` a constant?

**Comment by skriss** [comment]
@carlisia for some reason I had `else`'s on the brain this morning, idk why :) See latest commit.

File: pkg/controller/backup_controller.go [hunk 3]
```diff
@@ -398,6 +399,13 @@ func (c *backupController) runBackup(backup *pkgbackup.Request) error {
 	// Otherwise, the JSON file in object storage has a CompletionTimestamp of 'null'.
 	backup.Status.CompletionTimestamp.Time = c.clock.Now()
 
+	backup.Status.VolumeSnapshotsAttempted = len(backup.VolumeSnapshots)
```
Comment (nrb) [inline pkg/controller/backup_controller.go]: Any interest in sending these numbers to prometheus, aggregated by schedule?

Comment (skriss) [reply pkg/controller/backup_controller.go]: @nrb how about I add a separate issue for it and we leave this as-is?

**Pull request merged**